### PR TITLE
[HIPIFY][Complex] Add cuComplex support

### PR DESCRIPTION
**Pull request by emankov**
+ Add API_COMPLEX support (data types and functions)
+ Add cuComplex_API_supported_by_HIP.md
+ Add cuComplex_Julia.cu test
+ Update README.md

### CODE REVIEW

File: hipify-clang/src/Statistics.cpp [hunk 1]
```diff
@@ -8,13 +8,13 @@ const char *counterNames[NUM_CONV_TYPES] = {
...
     "cache", "exec", "err", "def", "tex", "gl", "graphics",
-    "surface", "jit", "d3d9", "d3d10", "d3d11", "vdpau", "egl",
+    "surface", "jit", "d3d9", "d3d10", "d3d11", "vdpau", "egl", "complex",
     "thread", "other", "include", "include_cuda_main_header", "type", "literal",
     "numeric_literal"
 };
 
 const char *apiNames[NUM_API_TYPES] = {
-    "CUDA Driver API", "CUDA RT API", "CUBLAS API", "CURAND API", "CUDNN API", "CUFFT API"
+    "CUDA Driver API", "CUDA RT API", "CUBLAS API", "CURAND API", "CUDNN API", "CUFFT API", "cuComplex API"
```
Comment (dfukalov) [inline hipify-clang/src/Statistics.cpp]: why this api name is not capitalized as other?

File: hipify-clang/src/CUDA2HipMap.cpp [hunk 2]
```diff
@@ -397,12 +396,11 @@ const std::map <llvm::StringRef, hipCounter> CUDA_INCLUDE_MAP{
     // CUDNN includes
     {"cudnn.h",                   {"hipDNN.h",                 CONV_INCLUDE_CUDA_MAIN_H, API_DNN}},
 
-    // CUDNN includes
+    // CUFFT includes
     {"cufft.h",                   {"hipfft.h",                 CONV_INCLUDE_CUDA_MAIN_H, API_FFT}},
 
-    // HIP includes
-    // TODO: uncomment this when hip/cudacommon.h will be renamed to hip/hipcommon.h
```
Comment (dfukalov) [inline hipify-clang/src/CUDA2HipMap.cpp]: do we need this todo?

File: hipify-clang/src/Statistics.cpp (hunk 1 above)
Comment (emankov) [reply hipify-clang/src/Statistics.cpp]: May be because Complex is not an abbreviation like Driver.

File: hipify-clang/src/CUDA2HipMap.cpp (hunk 2 above)
Comment (emankov) [reply hipify-clang/src/CUDA2HipMap.cpp]: No, we don't. It is appeared with SHOC hipification which now works fine with included cudacommon.h.

**Pull request merged**